hbase-server/src/test/java/org/apache/hadoop/hbase/util/TestFSTableDescriptors.java

```diff
@@ -285,6 +285,25 @@ public void testGetAll() throws IOException, InterruptedException {
       + htds.getAll().size(), count + 1, htds.getAll().size());
   }
 
+  @Test
+  public void testGetAllByParallel() throws IOException, InterruptedException {
+    final String name = "testGetAllByParallel";
+    FileSystem fs = FileSystem.get(UTIL.getConfiguration());
+    // Enable parallel load table descriptor.
+    FSTableDescriptors htds = new FSTableDescriptorsTest(fs, testDir, true, 20);
+    final int count = 100;
+    // Write out table infos.
+    for (int i = 0; i < count; i++) {
+      htds.createTableDescriptor(
+        TableDescriptorBuilder.newBuilder(TableName.valueOf(name + i)).build());
+    }
+    // add hbase:meta
+    htds
+      .createTableDescriptor(TableDescriptorBuilder.newBuilder(TableName.META_TABLE_NAME).build());
+    assertEquals("getAll() didn't return all TableDescriptors, expected: " + (count + 1) + " got: "
+      + htds.getAll().size(), count + 1, htds.getAll().size());
```

Comment:
```suggestion
    int getTableDescriptorSize = htds.getAll().size();
    assertEquals("getAll() didn't return all TableDescriptors, expected: " + (count + 1) + " got: "
      + getTableDescriptorSize, count + 1, getTableDescriptorSize);
    // get again to check whether the cache works well
    getTableDescriptorSize = htds.getAll().size();
    assertEquals("getAll() didn't return all TableDescriptors with cache, expected: " + (count + 1)
      + " got: " + getTableDescriptorSize, count + 1, getTableDescriptorSize);
```
We also need to check whether the cache works well.